src/showingpreviously/cinemas/glasgow_film_theatre.py

```diff
@@ -0,0 +1,142 @@
+                        json_attributes={**film_attributes, **showing_attributes}
+                    )
+                    showings.append(showing)
+        return showings
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def parse_film_title(film_title: str) -> (str, dict[str, any]):
+    # use lowercase
+    film_title = film_title.lower()
+    film_attributes = {'format': []}
+    # strip the rating from the film
+    # note: the rating is always last
+    if film_title.find('(') != -1:
+        film_title = film_title[:film_title.find('(')]
+    # check for ignore keywords
+    for keyword in FILM_TITLE_IGNORES:
+        if keyword.lower() in film_title:
+            film_title = film_title.replace(keyword.lower(), '')
+    # check for 4K screenings
+    if '4k' in film_title:
+        film_attributes['format'] += ['digital', '4K']
+        film_title = film_title.replace('4k', '').strip()
+    # check for 35mm screenings
+    if '35mm' in film_title:
+        film_attributes['format'].append('35mm')
+        film_title = film_title.replace('35mm', '').strip()
+    # if no format is given, assume digital
+    if len(film_attributes['format']) == 0:
+        film_attributes['format'] = ['digital']
+    # clean up the title by removing trailing characters,
+    # blank spaces and double-spaces
+    while True:
+        # assume no changes
+        old_title = film_title
+        # remove any trailing '+' or '-'
+        if film_title.strip().endswith('-') or film_title.strip().endswith('+'):
+            film_title = film_title.strip()[:-1]
+        # remove any double-spaces from removing two items from the title
+        film_title = film_title.replace('  ', ' ').strip()
+        if old_title == film_title:
+            break
+    return film_title, film_attributes
+    
+
+def film_from_heading(film_heading: bs4.element.Tag) -> (Film, dict[str, any]):
+    global films_cache
+    film_title, film_attributes = parse_film_title(film_heading.find('h3').text)
+    if film_title in films_cache:
+        return films_cache[film_title]
+    film = None
+    film_details_page_url = f'{BASE_URL}{film_heading.find("h3").find("a")["href"]}'
+    film_details_page = get_response(film_details_page_url).text
+    soup = bs4.BeautifulSoup(film_details_page, 'html.parser')
+    try:
+        metadata = soup.select('ul.inline-grid-row.meta')[0]
+    except IndexError:
+        film = Film(name=film_title, year=UNKNOWN_FILM_YEAR)
+        films_cache[film_title] = film, {}
+        return film, {}
+    for meta_item in metadata.find_all('li'):
+        meta_label = meta_item.find('span', class_='meta-label').text.strip()
+        meta_value = meta_item.find('span', class_='meta-value').text.strip()
+        if meta_label == 'Year of Production':
```

Comment:
For some films (like the [noirvember ones](https://glasgowfilm.org/shows/noirvember-bound-introduction-18)) the label is simply `Year`, not `Year of Production` - can we handle both cases please?